src/library/dao/trackdao.cpp

```diff
@@ -745,13 +741,13 @@ QList<TrackId> TrackDAO::addTracks(const QList<QFileInfo>& fileInfoList,
         int addIndex = query.value(addIndexColumn).toInt();
         QString filePath = query.value(locationColumn).toString();
         const QFileInfo& fileInfo = fileInfoList.at(addIndex);
-        TrackInfoObject* pTrack = new TrackInfoObject(fileInfo);
-        addTracksAdd(pTrack, unremove);
+        TrackPointer pTrack(TrackInfoObject::newTemporaryForFile(fileInfo));
```

Comment:
After looking around a bit, I think it is correct to use "newTemporaryForFile" here because of speed. 
Do you see a chance to loose meta-data changes in other temporary instances?  I don't. Since this is used for mass import. 

Please verify this and place a comment. Thank you.